lib/bolt/outputter.rb

```diff
@@ -14,6 +14,15 @@ def self.for_format(format)
     def initialize(stream = $stdout)
       @stream = stream
     end
+
+    def replace_undef(params)
```

Comment:
This could use a comment explaining what it's doing and why. Also it might be more appropriately named `replace_data` or `replace_data_type`. Technically it will be Data when it was undefined, but it can also be set to that explicitly; also, `undef` has very specific meaning in puppet types and doesn't apply here.